tests/test_AggregateConnectionData.cpp

```diff
@@ -0,0 +1,719 @@
+        auto xw = ::Opm::data::WellRates{};
+
+        {
+            xw["PROD"].rates
+                .set(o::wat, 1.0)
+                .set(o::oil, 2.0)
+                .set(o::gas, 3.0);
+	    xw["PROD"].bhp = 213.0;
+	    double qo = 5.;
+	    double qw = 4.;
+	    double qg = 50.;
+            for (int i = 0; i < 5; i++) {
+		xw["PROD"].connections.emplace_back();
+		auto& c = xw["PROD"].connections.back();
+		
+		c.rates.set(o::wat, qw*(float(i)+1.))
+                   .set(o::oil, qo*(float(i)+1.))
+                   .set(o::gas, qg*(float(i)+1.));
+		c.pressure = 215.; 
+	    }
+	    auto seg = Opm::data::Segment{};
+	    for (std::size_t i = 1; i < 5; i++) {
+		xw["PROD"].segments.insert(std::pair<std::size_t,Opm::data::Segment>(i,seg));
+	    }
+            xw["WINJ"].bhp = 234.0;
+
+            xw["WINJ"].rates.set(o::wat, 5.0);
+	    xw["WINJ"].rates.set(o::oil, 0.0);
+	    xw["WINJ"].rates.set(o::gas, 0.0);
+	    qw = 7.;
+            for (int i = 0; i < 5; i++) {
+		xw["WINJ"].connections.emplace_back();
+		auto& c = xw["WINJ"].connections.back();
+		
+		c.rates.set(o::wat, qw*(float(i)+1.))
+		       .set(o::oil, 0.)
+		       .set(o::gas, 0.);
+		c.pressure = 218.;
+	    }
+        }
+        return xw;
+    }
+}
+
+struct SimulationCase
+{
+    explicit SimulationCase(const Opm::Deck& deck)
+        : es   ( deck )
+        , grid(deck)
+        , sched( deck, es )
+    {}
+
+    // Order requirement: 'es' must be declared/initialised before 'sched'.
+    Opm::EclipseState es;
+    Opm::EclipseGrid grid;
+    Opm::Schedule     sched;
+};
+
+// =====================================================================
+
+BOOST_AUTO_TEST_SUITE(Aggregate_ConnData)
+
+
+// test dimensions of Connection data
+BOOST_AUTO_TEST_CASE (Constructor)
+{
+    const auto ih = MockIH{ 5 };
+   
+    const auto amconn = Opm::RestartIO::Helpers::AggregateConnectionData{ ih.value };
+
+    BOOST_CHECK_EQUAL(amconn.getIConn().size(), ih.nwells * ih.ncwmax * ih.niconz);
+    BOOST_CHECK_EQUAL(amconn.getSConn().size(), ih.nwells * ih.ncwmax * ih.nsconz);
+    BOOST_CHECK_EQUAL(amconn.getXConn().size(), ih.nwells * ih.ncwmax * ih.nxconz);
+}
+
+
+BOOST_AUTO_TEST_CASE (Declared_Connection_Data)
+{
+    const auto simCase = SimulationCase{first_sim()};
+
+    // Report Step 1: 2115-01-01 --> 2015-01-03
+    const auto rptStep = std::size_t{1};
+
+    const auto ih = MockIH {
+        static_cast<int>(simCase.sched.getWells(rptStep).size())
+    };
+
+    BOOST_CHECK_EQUAL(ih.nwells, MockIH::Sz{2});
+
+    const Opm::data::WellRates wrc = wr();
+    auto amconn = Opm::RestartIO::Helpers::AggregateConnectionData{ih.value};
+    amconn.captureDeclaredConnData(simCase.sched,
+				  simCase.grid,
+				  simCase.es.getUnits(), 
+				  wrc,
+				  rptStep
+				  );
+
+    // ICONN (PROD)
+    {
+        auto start = 0*ih.niconz;
+
+        const auto& iConn = amconn.getIConn();
+	BOOST_CHECK_EQUAL(iConn[start +  0] ,  1); // PROD-connection 1, sequence number
```

Comment:
The number `'a'` in `iConn[start + a]` *really* should have a name somewhere.

Reply:
> The number `'a'` in `iConn[start + a]` _really_ should have a name somewhere.

It [does](https://github.com/OPM/opm-common/blob/d9c81da18ce61c7034109a3112ed5629e42ec2ad/opm/output/eclipse/VectorItems/connection.hpp#L26-L27).  Using an explicit numeric index also verifies that the named indices in connection.hpp really do contain what they're supposed to.